ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/services/CommandService.scala

```diff
@@ -18,13 +26,47 @@ class CommandService(channel: Channel, authorizationToken: Option[String]) {
   private val service: CommandServiceGrpc.CommandServiceStub =
     AuthorizationHelper.maybeAuthedService(authorizationToken)(CommandServiceGrpc.stub(channel))
 
-  def submitAndWait(commands: Commands)(implicit ec: ExecutionContext): Future[Empty] =
+  def submitAndWaitForTransactionTree(
+      commands: Commands
+  )(implicit ec: ExecutionContext): Future[Seq[ScriptLedgerClient.CommandResult]] =
     service
-      .submitAndWait(new SubmitAndWaitRequest(Some(commands)))
+      .submitAndWaitForTransactionTree(new SubmitAndWaitRequest(Some(commands)))
```

Comment:
Why is it necessary to have a two step submission process?

Reply:
Have you tried to use createAndExercise construction? It is a way of stacking transactions into a larger one.

Reply:
`createAndExercise` let us do a single create followed by a single exercise. However we can generate more than one nonconsuming exercise followed by at most one consuming exercise for each create event. That's why I'm using this two step process.

An alternative would be to implement the generation and submission of nonconsuming and consuming exercises in Daml itself but has its own challenges as it introduces new auxilliary event(s) which:
- might change the characteristics of the data in index db beyond what was specific in the BenchTool's submission config.
- would have to accept random payloads for all consequence events or generate those payloads in Daml

Reply:
Seconding Marton's question: Why can't we submit all commands as part of a single submission? Does it matter for this benchmark whether they appear in separate transactions in the transactions stream?

Reply:
If the problem is the contractId reference that's not known for exercises, can we use key the contracts and `ExerciseByKey` instead?

Reply:
For the record, implemented `ExcerciseByKey` approach